### DISCUSSION TITLE
how to edit message of dialogue popup when input()

### DISCUSSION THREAD

**Discussion by tony-jinwoo-ahn** [pyodide/pyodide#3785] [Q&A] [answered]
when i run code like this:
```python
a=input()
```

I can see dialogue popup for stdin input,
and I'd like to edit message of that dialogue popup.
how can i modify the alert message?
Where can I modify in pyodide src code?

**Comment by hoodmane** [comment]
You can use`pyodide.setStdin()` for this.

**Reply by tony-jinwoo-ahn** [reply]
Thank you, but how can I use pyodide.setstdin() to change pop-up dialogue message?
I've read https://pyodide.org/en/stable/usage/api/js-api.html#pyodide.setStdin, but cannot find about custom message.

**Reply by tony-jinwoo-ahn** [reply]
@hoodmane 

I tested:
```
pyodide.setStdin('Hello! This is input message!');
```

I got:
```
OSError: [Errno 29] I/O error
```

I think there is a correct way, but can't find it.

**Reply by ryanking13** [reply, marked as answer]
@tony-jinwoo-ahn 

You need to pass the handler function, for example:

```js
pyodide.setStdin({stdin: () => prompt("Default Message")})
```

**Reply by tony-jinwoo-ahn** [reply]
![image](https://user-images.githubusercontent.com/43518162/233977628-579dee6e-72f1-4fa1-8293-28c95a4e4f45.png)

@ryanking13 Thank you, I can see "Default Message", but I'd like to change the "website says" message of dialogue. Maybe it's not related to Pyodide? I'm now searching about browser features.

**Reply by ryanking13** [reply]
>  Maybe it's not related to Pyodide? I'm now searching about browser features.

I think you need to implement a custom prompt for that. Then the problem is that since it is not possible to block the browser's main thread, you need to do it with a webworker. [This video](https://youtu.be/-SggWFS15Do) should help.

**Reply by tony-jinwoo-ahn** [reply]
@ryanking13 Thank you! It would be really helpful!
